service/worker/scheduler/workflow.go

```diff
@@ -151,13 +158,14 @@ func SchedulerWorkflow(ctx workflow.Context, args *schedspb.StartScheduleArgs) e
 		StartScheduleArgs: *args,
 		ctx:               ctx,
 		a:                 nil,
-		logger:            sdklog.With(workflow.GetLogger(ctx), "schedule-id", args.State.ScheduleId),
+		logger:            sdklog.With(workflow.GetLogger(ctx), "wf-namespace", args.State.Namespace, "schedule-id", args.State.ScheduleId),
+		metrics:           workflow.GetMetricsHandler(ctx).WithTags(map[string]string{"namespace": args.State.Namespace}),
```

Comment:
Do you need to set the metrics handler in per ns worker? https://github.com/temporalio/temporal/blob/7e89af4820f800427dacabbdee450c68531cfcf0/service/worker/pernamespaceworker.go#L434-L437

Reply:
The client factory set it (plus logger, hostport, etc.)